library/iFixit/Matryoshka/McRouter.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace iFixit\Matryoshka;
+
+use iFixit\Matryoshka;
+
+class McRouter extends Memcached {
+   const MCROUTER_INVALID_CHARACTERS = " \r\n";
+
+   /**
+    * Override the Memcahed factory method. Fail hard if binary protocol is
+    * set on the \Memcached backend. Mcrouter only supports ASCII protocol.
+    */
+   public static function create(\Memcached $memcached) {
+      if ($memcached->getOption(\Memcached::OPT_BINARY_PROTOCOL)) {
+         throw new \InvalidArgumentException("Binary Protocol is not supported");
+      }
+
+      self::setGetMultiParams($memcached);
+
+      return new KeyFix(new self($memcached),
+       self::MAX_KEY_LENGTH, self::MCROUTER_INVALID_CHARACTERS);
+   }
+
+   /**
+    * Override the `set` method. Use `add` which is synchronous to detect
+    * `set` over-top of existing keys. Delete and reset them to
+    * enforce consistency.
+    */
+   public function set($key, $value, $expiration = 0) {
```

Comment:
`: bool` hinting

Reply:
The interface doesn't have type-hinting so we can't add it here without adding it everywhere.

Reply:
That's correct, I remember darryl explaining that now